### PR TITLE
Rename Testing guide to Testing with RSpec

### PR DESCRIPTION
**Pull request by WilHall**
Our testing guide was RSpec specific. This change renames it from `testing` to `testing-rspec` and rolling the best practices into the guide. This is the first step towards #611 where we can have a general testing guide as well as a guide for each framework.

### CODE REVIEW

File: best-practices/README.md [hunk 1]
```diff
@@ -105,38 +105,6 @@ Rails
...
-* Avoid `any_instance` in rspec-mocks and mocha. Prefer [dependency injection].
-* Avoid `its`, `specify`, and `before` in RSpec.
-* Avoid `let` (or `let!`) in RSpec. Prefer extracting helper methods,
-  but do not re-implement the functionality of `let`. [Example][avoid-let].
-* Avoid using `subject` explicitly *inside of an* RSpec `it` block.
-  [Example][subject-example].
-* Avoid using instance variables in tests.
-* Disable real HTTP requests to external services with
-  `WebMock.disable_net_connect!`.
-* Don't test private methods.
```
Comment (mike-burns) [inline best-practices/README.md:120]: Can we keep this one?

File: best-practices/README.md [hunk 2]
```diff
@@ -105,38 +105,6 @@ Rails
...
-* Test background jobs with a [`Delayed::Job` matcher].
-* Use [stubs and spies] \(not mocks\) in isolated tests.
-* Use a single level of abstraction within scenarios.
-* Use an `it` example or test method for each execution path through the method.
```
Comment (mike-burns) [inline best-practices/README.md:124]: Should we keep this one?

File: best-practices/README.md [hunk 3]
```diff
@@ -105,38 +105,6 @@ Rails
...
-* Use [assertions about state] for incoming messages.
-* Use stubs and spies to assert you sent outgoing messages.
-* Use a [Fake] to stub requests to external services.
```
Comment (mike-burns) [inline best-practices/README.md:127]: These last three seem generic enough.

**Comment by WilHall** [comment]
@mike-burns I just saw your comments on keeping some of the guidelines.

In this PR, I have not removed any; only renamed and formatted the document.

**Comment by mike-burns** [comment]
Oh! Sorry, I misread. My fault.

This looks great to me!

**Pull request merged**